### PR TITLE
[Improvement][worker] Optimize KillYarnJob (4939)

### LINKED ISSUE #4939: [Improvement][worker] Optimize KillYarnJob
When KillYarnJob, determine whether the target host is the local machine, if it is, rpc communication is not required

org.apache.dolphinscheduler.server.worker.processor.TaskKillProcessor#killYarnJob

### PR DESCRIPTION
**Pull request by ruanwenjun**
When we kill a yarn job, we need to find the application Id from the log content. If the log is store on the local machine, we can read it directly.

this closes #4939

### CODE REVIEW

**Comment by CalvinKirs** [comment]
hi, you need to add UT, the current UT coverage is not enough

**Comment by ruanwenjun** [comment]
> hi, you need to add UT, the current UT coverage is not enough

@CalvinKirs Thank you for your suggestion, I have added unit tests to cover the submitted code, but the coverage is still unqualified. How can I solve this problem?

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/LoggerUtils.java [hunk 1]
```diff
@@ -101,6 +108,26 @@ public static String buildTaskId(String affix,
...
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(filePath)))) {
+            while ((line = br.readLine()) != null) {
+                sb.append(line + "\r\n");
+            }
+            return sb.toString();
+        } catch (IOException e) {
+            logger.error("read file error", e);
+        }
+        return "";
+    }
```
Comment (CalvinKirs) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/LoggerUtils.java:129]: Did you forget to close the stream?

**Review by CalvinKirs — CHANGES_REQUESTED** [review]

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/LoggerUtils.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/LoggerUtils.java:129]: Due to the BufferedReader inherits AutoCloseable, so sonor suggest us to use try with resources to close the stream.

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**